### ISSUE TITLE
Support both `dotnet_diagnostic.` and `dotnet_code_quality.` prefixes

### ISSUE THREAD

**Issue by Evangelink** [dotnet/roslyn-analyzers#5742]
@shuebner However, you raise a good point, which @gewarren, @Youssef1313 and @Evangelink have also raised in the past. We should probably support both `dotnet_diagnostic.` prefix and `dotnet_code_quality.` prefix for specifying editorconfig options for CA rules (latter for backwards compatibility). Many customers have got confused due to different prefix for CA rule options. It's likely a very simple fix to the below method to add this support: https://github.com/dotnet/roslyn-analyzers/blob/186404d87db29c9a9cff55708289ae94ab5acaae/src/Utilities/Compiler/Options/AbstractCategorizedAnalyzerConfigOptions.cs#L47-L63

_Originally posted by @mavasani in https://github.com/dotnet/roslyn-analyzers/issues/5664#issuecomment-948495890_

**Comment by Evangelink** [comment]
@mavasani It appears that this method is unused.

**Comment by mavasani** [comment]
Tagging @jaredpar @chsienki, we need to first get the views + approval for the compiler team here, let me provide some context.

Our .NET SDK analyzers (CAxxxx rules) support a bunch of editorconfig options, which are documented [here](https://github.com/dotnet/roslyn-analyzers/blob/main/docs/Analyzer%20Configuration.md#editorconfig-format). Currently, all of these option keys require `dotnet_code_quality.` prefix. We have had bunch of customer reports where they accidentally ended up using `dotnet_diagnostic.` prefix for option entries, as they are used to `dotnet_diagnostic.RuleId.severity` syntax for setting diagnostic severity in editorconfig and they got confused why the option is not respected. We would like to consider switching our `CAxxxx` option entries to also have `dotnet_diagnostic.` prefix for consistency and ease of use. We will still support `dotnet_code_quality.` prefix for backwards compatibility, but switch all our documentation to `dotnet_diagnostic.` prefix. Does this sound reasonable to you?

Also tagging @gewarren for follow-up doc changes if we do decide to go ahead with this.

**Comment by mavasani** [comment]
> It appears that this method is unused.

@Evangelink Thanks. It seems we introduced an equivalent method below for the above unused method, which would need to be adjusted if we implement this feature support: https://github.com/dotnet/roslyn-analyzers/blob/186404d87db29c9a9cff55708289ae94ab5acaae/src/Utilities/Compiler/Options/AbstractCategorizedAnalyzerConfigOptions.cs#L39-L45

**Comment by chsienki** [comment]
@mavasani Seems fine to me. Just to confirm that we're happy with the prefix being the same, but the parts after it varying: `.RuleId.Severity` vs `.RuleId.OptionName` would presumably have no affect if applied to wrong kind of rule?

**Comment by mavasani** [comment]
> parts after it varying: .RuleId.Severity vs .RuleId.OptionName would presumably have no affect if applied to wrong kind of rule

Yes, that is correct. @Evangelink Feel free to take this up if you are still interested.

**Comment by Evangelink** [comment]
I was starting to work on this when I just clicked and now, I am wondering if it wouldn't be better to implement an analyzer to validate configurations. I guess that my still be worth to implement as I suspect the analyzer to work only in a limited context.

I will still provide the PR for that but thought it might be worth discussing/pointing this out.

**Comment by Evangelink** [comment]
FYI, it's taking me some time to create the PR because I am doing a refactoring on some of the `InlineData` to replace them with generators `MemberData` so that we can easily reuse testing the various cases (we currently do manual copy-paste).
